nemo_text_processing/g2p/modules.py

```diff
@@ -315,40 +316,29 @@ def __init__(
         else:
             self.use_chars = use_chars
 
+        # parse the dictionary, update it, and generate symbol set.
         if isinstance(phoneme_dict, str) or isinstance(phoneme_dict, pathlib.Path):
-            # Load phoneme_dict from file path
-            self.phoneme_dict, self.symbols = self._parse_as_cmu_dict(phoneme_dict)
+            # load the dictionary file where there may exist a digit suffix after a word, which
+            # represents the pronunciation variant of that word.
+            phoneme_dict_obj = defaultdict(list)
+            _alt_re = re.compile(r'\([0-9]+\)')
+            with open(phoneme_dict, "r") as fdict:
+                for line in fdict:
+                    if len(line) and ('A' <= line[0] <= 'Z' or line[0] == "'"):
+                        parts = line.strip().split()
+                        assert len(parts) == 2, f"Wrong format for the entry: {line.strip()}."
+                        word = re.sub(_alt_re, '', parts[0])
+                        phoneme_dict_obj[word].append(list(parts[1]))
         else:
             # Load phoneme_dict as dictionary object
             logging.info("Loading phoneme_dict as a Dict object. Extracting valid symbols from values.")
-            self.phoneme_dict = phoneme_dict
-
-            self.symbols = set()
-            for word, phonemes in phoneme_dict.items():
-                self.symbols.update(phonemes)
-                if self.use_chars:
-                    word_no_punct = self.PUNCT_REGEX.sub('', word)
-                    self.symbols.update(word_no_punct)
-
-            # Update dict to remove stresses if use_stresses=False
-            if not use_stresses:
-                for stress_symbol in self.STRESS_SYMBOLS:
-                    self.symbols.remove(stress_symbol)
-
-                updated_phoneme_dict = {}
-                for k, vs in self.phoneme_dict.items():
-                    new_vs = []
-                    for pron in vs:
-                        new_vs.append([symbol for symbol in pron if symbol not in self.STRESS_SYMBOLS])
-                    updated_phoneme_dict[k] = new_vs
-                self.phoneme_dict = updated_phoneme_dict
-
-            # Update dict keys if graphemes should be set to uppercase
-            if set_graphemes_upper:
-                updated_phoneme_dict = {}
-                for k, vs, in self.phoneme_dict.items():
-                    updated_phoneme_dict[k.upper()] = vs
-                self.phoneme_dict = updated_phoneme_dict
+            phoneme_dict_obj = phoneme_dict
+
+        # verify if phoneme dict obj is empty
+        if phoneme_dict_obj:
+            self.phoneme_dict, self.symbols = self._parse_as_cmu_dict(phoneme_dict_obj)
+        else:
+            raise ValueError(f"Found empty entries in {phoneme_dict}.")
```

Comment:
Would it be more accurate to say "phoneme_dict is empty" in this error message?

Reply:
Not really. `phoneme_dict` could be a `dict` or a file. it is fine to say that if input is a dict, but not good if the input is a file because the dict file could contain no entries but contain comments or license or something else. So it is not empty.

Reply:
Then should we say "phoneme_dict contains no entries"? The current text reads to me like there are word entries without pronunciations rather than no entries at all.

Reply:
Yes, that sounds much better! will revise.